Update gsheets.md (#5229)
dbt-labs/docs.getdbt.com: website/docs/docs/use-dbt-semantic-layer/gsheets.md

When querying your data with Google Sheets:

- It returns the data to the cell you have clicked on, and each cell where data is requested will have a note attached to it, indicating what has been queried and the timestamp.
- The custom menu operation has a timeout limit of six (6) minutes.
- If you're using this extension, make sure you're signed into Chrome with the same Google profile you used to set up the Add-On. Log in with one Google profile at a time as using multiple Google profiles at once might cause issues.

  
## Query Builder functions

> The custom menu provides the following capabilities:

Reviewer: ```suggestion
The **Query Builder** custom menu has the following capabilities:
```

> The **Query Builder** custom menu has the following capabilities:

Reviewer: ```suggestion
The Google Sheets **Query Builder** custom menu has the following capabilities:
```

| Metrics       | Search and select metrics.                             |
| Group By      | Search and select dimensions or entities to group by. Dimensions are grouped by the entity of the semantic model they come from. |
| Time Range    | Quickly select time ranges to look at the data, which applies to the main time series for the metrics (metric time). You can still optionally use `where` to filter time ranges|
| Granularity   | Modify the granularity of the primary time dimension.      |
| Where         | Filter your data. This includes categorical and time filters. |
| Order By      | Return your data order.                              |
| Limit         | Set a limit for the rows of your output.               |

**Filtering data**

To use the filter functionality, choose the [dimension](docs/build/dimensions) you want to filter by and select the operation you want to filter on. 
   - For categorical dimensions, type in the dimension value you want to filter by (no quotes needed) and press enter. 
   - Continue adding additional filters as needed with AND and OR. If it's a time dimension, choose the operator and select from the calendar.

> **Querying without Headers/Column**

Reviewer: ```suggestion
**Querying without headers or columns**
```

> **Querying without headers or columns**

Reviewer: @rpourzand do you think we need to have a 'columns' as well? wondering if just having 'headers' makes sense?

> If you would like to just query the data values without the headers, you can optionally check the box "Exclude Column Names".

Reviewer: fyi @rpourzand i dont see this right now in sheets - is that ok? might be helpful to have a screenshot to show what this looks like. non blocker

> If you would like to just query the data values without the headers, you can optionally check the box "Exclude Column Names".

Reviewer: ```suggestion
If you would like to just query the data values without the headers, you can optionally select the **Exclude Column Names** box.
```

> ## Creating and Accessing Saved selections

Reviewer: ```suggestion
## Creating and accessing Saved selections
```

> Saved selections allow you to save the inputs you've created in the Google Sheets **Query Builder** and easily access them again so you don't have to continuously build common queries from scratch. To create a saved selection, run a query and choose the option to save selection.

Reviewer: bolding the 'saved selection' option since i assume it's a gsheets UI and also adding the instruction to a separate line

```suggestion
Saved selections allow you to save the inputs you've created in the Google Sheets **Query Builder** and easily access them again so you don't have to continuously build common queries from scratch. 

To create a saved selection, run a query with the **Query Builder** and select the **Save Selection** option.
```

Reviewer: @rpourzand confirming it's 'Saved Selections' or 'Saved selections' in the UI?

The Google Sheets application saves these selections, allowing you to view and edit them from the hamburger menu under **Saved Selections**. You can also make these selections private or public:

- Public selections mean your inputs are available in the menu to everyone on the sheet.
- Private selections mean your inputs are only visible to you. Note that anyone added to the sheet can still see the data from these private selections, but they won't be able to interact with the selection in the menu, or benefit from the automatic refresh.

> ## Refreshing Selections

Reviewer: ```suggestion
## Refreshing selections
```

> ## Refreshing selections

Reviewer: ```suggestion
### Refreshing selections
```

> You can also refresh saved selections on load. When creating the saved selection option, choose the option to "Refresh on Load", which will refresh the data automatically when the addon is accessed after. For public saved selections, they will be refreshed for any editor of the sheet. For private selections, those will only be refreshed for the user who created it. 

Reviewer: ```suggestion
Set your saved selections to automatically refresh every time you load the addon. You can do this by selecting **Refresh on Load** when creating the saved selection. 

Public saved selections will refresh for anyone who edits the sheet, while private selections will only update for the user who created it. 
```

## Using Saved Queries in Google Sheets
Access [saved queries](/docs/build/saved-queries), powered by MetricFlow, in Google Sheets to quickly get results from pre-defined sets of data. 

To access the saved queries in Google Sheets:
1. Open the hamburger menu in Google Sheets.
2. Navigate to "Saved Queries" to access the ones available to you. 

> 3. You can also select **Customize**, which allows you to explore from the existing query. This won't change the original query that's defined in the code. 

Reviewer 2: ```suggestion
3. You can also select **Build Selection**, which allows you to explore from the existing query. This won't change the original query that's defined in the code. 
```

The Google Sheets application saves these selections, allowing you to view and edit them from the hamburger menu under **Saved Selections**. You can also make these selections private or public:

- Public selections mean your inputs are available in the menu to everyone on the sheet.
- Private selections mean your inputs are only visible to you. Note that anyone added to the sheet can still see the data from these private selections, but they won't be able to interact with the selection in the menu, or benefit from the automatic refresh.

### Refreshing selections

Set your saved selections to automatically refresh every time you load the addon. You can do this by selecting **Refresh on Load** when creating the saved selection. 

Public saved selections will refresh for anyone who edits the sheet, while private selections will only update for the user who created it. 

## Using Saved queries
Access [saved queries](/docs/build/saved-queries), powered by MetricFlow, in Google Sheets to quickly get results from pre-defined sets of data. 

To access the saved queries in Google Sheets:
1. Open the hamburger menu in Google Sheets.

> 2. Navigate to "Saved Queries" to access the ones available to you. 

Reviewer: ```suggestion
2. Navigate to **Saved Queries** to access the ones available to you. 
```

- Continue adding additional filters as needed with AND and OR. If it's a time dimension, choose the operator and select from the calendar.

**Querying without headers or columns**

If you would like to just query the data values without the headers, you can optionally select the **Exclude Column Names** box.

## Creating and accessing Saved selections
Saved selections allow you to save the inputs you've created in the Google Sheets **Query Builder** and easily access them again so you don't have to continuously build common queries from scratch. To create a saved selection, run a query and choose the option to save the selection.

The Google Sheets application saves these selections, allowing you to view and edit them from the hamburger menu under **Saved Selections**. You can also make these selections private or public:

- Public selections mean your inputs are available in the menu to everyone on the sheet.
- Private selections mean your inputs are only visible to you. Note that anyone added to the sheet can still see the data from these private selections, but they won't be able to interact with the selection in the menu or benefit from the automatic refresh.

> 

Reviewer: is it a toggle?

| Metrics       | Search and select metrics.                             |
| Group By      | Search and select dimensions or entities to group by. Dimensions are grouped by the entity of the semantic model they come from. |
| Time Range    | Quickly select time ranges to look at the data, which applies to the main time series for the metrics (metric time). You can still optionally use `where` to filter time ranges|
| Granularity   | Modify the granularity of the primary time dimension.      |
| Where         | Filter your data. This includes categorical and time filters. |
| Order By      | Return your data order.                              |
| Limit         | Set a limit for the rows of your output.               |

**Filtering data**

To use the filter functionality, choose the [dimension](docs/build/dimensions) you want to filter by and select the operation you want to filter on. 
   - For categorical dimensions, type in the dimension value you want to filter by (no quotes needed) and press enter. 
   - Continue adding additional filters as needed with AND and OR. If it's a time dimension, choose the operator and select from the calendar.

> ## Creating Saved Selections

Reviewer: ```suggestion
## Creating Saved selections
```

> Saved Selections allow you to save your inputs that you've created in the query builder and easily access them again so you don't have to continuously build common queries from scratch. These selections will be saved by your Google Sheets application and are viewable and editable from the hamburger menu under "Saved Selections". You can also make these selections private or public. Public selections will be available in the menu to everyone in the sheet whereas private ones will only be visible to you.

Reviewer: ```suggestion
Saved selections allow you to save the inputs you've created in the Google Sheets **Query Builder** and easily access them again so you don't have to continuously build common queries from scratch. 

The Google Sheets application saves these selections, allowing you to view and edit them from the hamburger menu under **Saved Selections**. You can also make these selections private or public:

- Public selections mean your inputs are available in the menu to everyone on the sheet.
- Private selections mean your inputs are only visible to you.
```

> ## Using Saved Queries in Google Sheets

Reviewer: ```suggestion
## Using Saved queries
```

> Saved Queries (add link) in MetricFlow can be accessed in Google Sheets to quickly get results from these pre-defined sets of data. In the hamburger menu, navigate to "Saved Queries" to access the ones available to you. You can also choose "Customize", which allows you to jump off and explore from the existing query, but it will never change the query as defined in code. Note that for these saved queries, if a WHERE filter is defined in the query, we will surface the advanced syntax. 

Reviewer: ```suggestion
Access [saved queries](/docs/build/saved-queries), powered by MetricFlow, in Google Sheets to quickly get results from pre-defined sets of data. 

To access the saved queries in Google Sheets:
1. Open the hamburger menu in Google Sheets.
2. Navigate to "Saved Queries" to access the ones available to you. 
3. You can also select **Customize**, which allows you to explore from the existing query. This won't change the original query that's defined in the code. 
  - For saved queries, if you use a `WHERE` filter in a query, Google Sheets dispalys the advanced syntax for this filter.
```

Reviewer: maybe a screenshot?

> maybe: Note that difference between using Saved Queries and Saved Selections? Saved query = use predefined in-code input to get result, jump off. Saved selection =  create sets of things that can be re-used in gooogle sheets in the app, not in code (this includes saving the results from a saved query)

Reviewer: @rpourzand silly question from me but  what does 'jump off' mean here? 🙈

```suggestion
<expandable alt_header="What's the difference between Saved queries and Saved selections?">

- Saved queries mean you use predefined code inputs to receive results and jump off.
- Saved selection means you create sets of things (inputs) that can be reused in the Google Sheets application, not in code. This includes saving the results from a saved query.

</expandable>
```

Author: Ah maybe we should use different language, not a silly question! Basically you can use the saved query as an explore point to "jump off" and build your own query
